### PR TITLE
Changes required to manage Katello modules.

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ekohl** [comment]
Looks good. Have you done any run to see how badly everything fails with the additional puppet-lint checks?

**Comment by ehelms** [comment]
All in all, it appears to be simple things to fix, the common issues I saw were:

1) Puppet 4 support: when Puppet 4 was installed they would fail with 

```
/home/ehelms/.rvm/gems/ruby-2.1.0/gems/puppetlabs_spec_helper-0.10.2/lib/puppetlabs_spec_helper/module_spec_helper.rb:23:in `block in <top (required)>': undefined method `environmentpath='
```

2) file_concat missing, which I was informed is us needing to add a new fixture

```
Invalid resource type file_concat
```

3) metadata breakages for our 'unreleased' modules, mostly the license name
4) Variable assigned to empty string; what's the right way to handle this?
5) Class included by relative name

**Comment by ekohl** [comment]
1) that's not yet supported (which IMHO is a bad thing for a released language)
4) that sort of depends on the situation. What we did was open a PR for every module and fix it case by case till Travis was happy

**Comment by ehelms** [comment]
@ekohl Puppet 4 is not supported or just the spec helper? Since the Gemspec does not limit Puppet to less than 4 at the moment, its not obvious to have to set PUPPET_VERSION before a bundle run locally at least.

**Comment by ehelms** [comment]
For the empty string issue, does it only care about parameters? For example, would https://github.com/Katello/puppet-qpid/blob/master/manifests/router/connector.pp#L21 just need to be turned into a params variable?

**Comment by ekohl** [comment]
It does care about that. The reasoning is that the following code behaves different in puppet 3 and puppet 4 since the latter evaluates empty strings to a trueish value:

``` puppet
$var = ''
if $var {
    # code
} else {
    # other code
}
```

**Comment by ehelms** [comment]
@stbenjam thoughts / review ?

Once / if we agree upon this as a starting place I'll work on updating our modules based on this.

**Comment by stbenjam** [comment]
Very cool, ACK from me

I guess it will generate some test failures? Looks like noop mode will dump the changes into the modules dir locally - what do you think about a manual run this first time so we can fix the test failures?

**Comment by ehelms** [comment]
I was only considering using this in a manual sense (in part because it was unclear to me how the 'automatic' case is managed).

**Comment by stbenjam** [comment]
"With no additional arguments, ModuleSync clones modules from the puppetlabs github organization and pushes to the master branch." Sounds like it just happily pushes directly by default? I didn't try it of course :)

**Comment by ehelms** [comment]
I suppose we could set a default branch so that if it is run it gets pushed to that instead of master.

**Comment by ehelms** [comment]
@stbenjam @ekohl Added a branch configuration for the case we talked about. Let me know what you think overall, as I'd like to merge this and then start updating modules based upon it. Note that I updated the Gemfile to restrict puppet so please look at that as well.

File: moduleroot/Gemfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 source 'https://rubygems.org'
 
-gem 'puppet', ENV.key?('PUPPET_VERSION') ? "~> #{ENV['PUPPET_VERSION']}" : '>= 2.7'
+gem 'puppet', ENV.key?('PUPPET_VERSION') ? "~> #{ENV['PUPPET_VERSION']}" : '>= 3.2'
```
Comment (ekohl) [inline moduleroot/Gemfile]: Since travis tests 3.5+ I'd keep those in sync.

**Comment by ekohl** [comment]
I think it looks good and I'd create the PRs and fix it module per module.

**Pull request merged**